Examples/Scripts/Python/full_chain_odd.py

```diff
@@ -1,60 +1,78 @@
 #!/usr/bin/env python3
 import pathlib, acts, acts.examples
-import acts.examples.dd4hep
-from common import getOpenDataDetectorDirectory
-from acts.examples.odd import getOpenDataDetector
-
-# acts.examples.dump_args_calls(locals())  # show python binding calls
-
-u = acts.UnitConstants
-outputDir = pathlib.Path.cwd() / "odd_output"
-outputDir.mkdir(exist_ok=True)
-
-oddDir = getOpenDataDetectorDirectory()
-
-oddMaterialMap = oddDir / "data/odd-material-maps.root"
-oddDigiConfig = oddDir / "config/odd-digi-smearing-config.json"
-oddSeedingSel = oddDir / "config/odd-seeding-config.json"
-oddMaterialDeco = acts.IMaterialDecorator.fromFile(oddMaterialMap)
-
-detector, trackingGeometry, decorators = getOpenDataDetector(
-    getOpenDataDetectorDirectory(), mdecorator=oddMaterialDeco
-)
-field = acts.ConstantBField(acts.Vector3(0.0, 0.0, 2.0 * u.T))
-rnd = acts.examples.RandomNumbers(seed=42)
-
 from acts.examples.simulation import (
     addParticleGun,
+    addPythia8,
     MomentumConfig,
     EtaConfig,
     ParticleConfig,
     addFatras,
     addDigitization,
+    ParticleSelectorConfig,
 )
 from acts.examples.reconstruction import (
     addSeeding,
     addCKFTracks,
     CKFPerformanceConfig,
     addVertexFitting,
     VertexFinder,
+    TruthSeedRanges,
 )
+from common import getOpenDataDetectorDirectory
+from acts.examples.odd import getOpenDataDetector
 
-s = acts.examples.Sequencer(events=100, numThreads=-1, logLevel=acts.logging.INFO)
+ttbar_pu200 = False
+u = acts.UnitConstants
+geoDir = getOpenDataDetectorDirectory()
+outputDir = pathlib.Path.cwd() / "odd_output"
+outputDir.mkdir(exist_ok=True)
+# acts.examples.dump_args_calls(locals())  # show python binding calls
 
-addParticleGun(
-    s,
-    MomentumConfig(1.0 * u.GeV, 10.0 * u.GeV, transverse=True),
-    EtaConfig(-3.0, 3.0, uniform=True),
-    ParticleConfig(2, acts.PdgParticle.eMuon, randomizeCharge=True),
-    rnd=rnd,
+oddMaterialMap = geoDir / "data/odd-material-maps.root"
+oddDigiConfig = geoDir / "config/odd-digi-smearing-config.json"
+oddSeedingSel = geoDir / "config/odd-seeding-config.json"
+oddMaterialDeco = acts.IMaterialDecorator.fromFile(oddMaterialMap)
+
+detector, trackingGeometry, decorators = getOpenDataDetector(
+    geoDir, mdecorator=oddMaterialDeco
 )
+field = acts.ConstantBField(acts.Vector3(0.0, 0.0, 2.0 * u.T))
+rnd = acts.examples.RandomNumbers(seed=42)
+
+s = acts.examples.Sequencer(events=100, numThreads=-1, logLevel=acts.logging.INFO)
+
+if not ttbar_pu200:
+    addParticleGun(
+        s,
+        MomentumConfig(1.0 * u.GeV, 10.0 * u.GeV, transverse=True),
+        EtaConfig(-4.0, 4.0, uniform=True),
+        ParticleConfig(2, acts.PdgParticle.eMuon, randomizeCharge=True),
+        rnd=rnd,
+    )
+else:
+    addPythia8(
+        s,
+        hardProcess=["Top:qqbar2ttbar=on"],
+        npileup=200,
+        vtxGen=acts.examples.GaussianVertexGenerator(
+            stddev=acts.Vector4(0.0125 * u.mm, 0.0125 * u.mm, 55.5 * u.mm, 5.0 * u.ns),
```

Comment:
This is the LHC beamspot and (maximum pile-up mu). I don't know what fantasy accelerator the ODD is for. Are these settings still OK?

Reply:
I think this should be fine